### PR TITLE
fix(metadata): change description from documentdb_cluster_deletion_protection

### LINKED ISSUE #4906: Incorrect Description in documentdb_cluster_deletion_protection.metadata.json for DocumentDB
### Steps to Reproduce

1. Navigate to the documentdb_cluster_deletion_protection.metadata.json file in the Prowler repository: https://github.com/prowler-cloud/prowler/blob/master/prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json

2.Review the description in line 11 of the file.
3.Notice the description states: "Check if Neptune Clusters has deletion protection enabled."

### Expected behavior

The description should accurately describe the check for DocumentDB clusters, not Neptune clusters. It should read:

"Check if DocumentDB Clusters has deletion protection enabled."

### Actual Result with Screenshots or Logs

Actual Description: "Check if Neptune Clusters has deletion protection enabled."
Corrected Description: "Check if DocumentDB Clusters has deletion protection enabled."

This appears to be a simple text entry error.
Upon checking, the check for Neptune clusters exists with the correct descriptions in both the latest master branch and the 4.3.5 release version (neptune_cluster_deletion_protection in 4.3.5 and neptune_cluster_deletion_protection_enabled in older versions).

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

My local workstation.

### OS used

1.Amazon linux and MacOS

### Prowler version

4.3.5(PYPL),4.4(git)

### Pip version

pip 24.2 

### Context

Additional context: This issue was identified by cross-referencing the metadata files for both DocumentDB and Neptune checks in the repository. The Neptune check descriptions are correct and do not need modifications.

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

Fix #4906

### Description

Change check metadata description from `documentdb_cluster_deletion_protection`
From : `Check if Neptune Clusters has deletion protection enabled.`
To: `Check if DocumentDB Clusters has deletion protection enabled.`

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "ResourceIdTemplate": "arn:aws:rds:region:account-id:db-cluster",
   "Severity": "medium",
   "ResourceType": "AwsRdsDbClusters",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json]: Is the `ResourceType` right?

File: prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
...
-  "Description": "Check if Neptune Clusters has deletion protection enabled.",
+  "Description": "Check if DocumentDB Clusters has deletion protection enabled.",
   "Risk": "Enabling cluster deletion protection offers an additional layer of protection against accidental database deletion or deletion by an unauthorized user. A Neptune DB cluster can't be deleted while deletion protection is enabled. You must first disable deletion protection before a delete request can succeed.",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json]: There are more references to Neptune here.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json (hunk 2 above)
Comment (baecheongho) [reply prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json]: As jfagoagas mentioned, there is also an incorrect reference to "Neptune" in the Risk field that should be updated to "DocumentDB."

File: prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json]: No, I've changed it

File: prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/aws/services/documentdb/documentdb_cluster_deletion_protection/documentdb_cluster_deletion_protection.metadata.json]: True, thanks

**Review by jfagoagas — APPROVED** [review]
Good catch!

**Pull request merged**